libtiny_client/src/state.rs

```diff
@@ -232,6 +271,31 @@ impl StateInner {
                 }
             }
 
+            // Reply 477 when user needs to be identified with NickServ to join a channel
+            Reply { num: 477, params } => {
+                // Only try to automatically rejoin if nickserv_ident is configured
+                if self.nickserv_ident.is_some() {
+                    // Get channel name
+                    let channel = params.get(1).unwrap().clone();
+                    // Find channel in self.chans
+                    let idx = utils::find_idx(&self.chans, |c| c.name == channel).unwrap();
```

Comment:
Let's also remove this unwrap which can cause a crash when we reconnect to a server (so the chan tab already exists) but the user closes the chan tab *after* sending JOIN but *before* receiving the JOIN response.